nixos/modules/services/networking/spacecookie.nix

```diff
@@ -0,0 +1,90 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.spacecookie;
+    defaultUser = "spacecookie";
+    configFile = pkgs.writeText "spacecookie.json" (lib.generators.toJSON {} {
+        hostname = cfg.hostname;
+        port = cfg.port;
+        user = cfg.user;
+        root = cfg.root;
+      });
+in {
+
+  options = {
+
+    services.spacecookie = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable the Spacecookie Gopher Daemon";
+      };
+
+      hostname = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = "The hostname the service is reachable via. Clients will use this hostname for further requests after loading the initial gopher menu.";
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 70;
+        description = "Port the gopher service should be exposed on.";
+      };
+
+      root = mkOption {
+        type = types.path;
+        default = "/srv/gopher";
+        description = "The root directory spacecookie serves via gopher.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "spacecookie";
+        description = "User the daemon should be run as.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.sockets.spacecookie = {
+      description = "Socket for the Spacecookie Gopher Server";
+      wantedBy = [ "sockets.target" ];
+      listenStreams = [ "[::]:${toString cfg.port}" ];
+      socketConfig.BindIPv6Only = "both";
+      socketConfig.SocketUser = cfg.user;
+      socketConfig.SocketGroup = cfg.user;
```

Comment:
`SocketMode` defaults to `0666` - is this what you want?

Reply:
No, good suggestion!